### PR TITLE
Refactor modules and update synthax

### PR DESCRIPTION
**Pull request by CaiqueMitsuoka**
A refactor to change the namespace of the business rules that was inside `Link` which is the repo structure and shouldnt have this kind of logic inside.

Also remove some data converter functions and to use the `Link` data structure instead of creating a similar map.
This helped to clean up most of the modules.

Does not include any visual change.

### CODE REVIEW

**Review by feliperli — APPROVED** [review]

File: lib/baelish/shortener/persistance.ex [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  end
+
+  defp persist(url) do
+    to_string(url)
+    |> check_existance()
+    |> insert_database(url)
+    |> insert_cache()
+  end
+
+  defp check_existance(url) do
```
Comment (elainenaomi) [inline lib/baelish/shortener/persistance.ex]: existence 🙂

File: lib/baelish/shortener/persistance.ex [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  def perform(url) do
+    persist(url)
+  end
+
+  defp persist(url) do
+    to_string(url)
+    |> check_existance()
+    |> insert_database(url)
+    |> insert_cache()
```
Comment (elainenaomi) [inline lib/baelish/shortener/persistance.ex]: Do you need to use parenthesis in this pipeline?  🙂

File: lib/baelish/shortener/persistance.ex [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  defp insert_database(%{link: link, uid: uid}, _) do
+    {:ok, link}
+  end
+
+  defp insert_cache({:ok, link}) do
+    Cache.create(link)
+  end
+
+  defp insert_cache({:error, _}) do
+    :error
```
Comment (elainenaomi) [inline lib/baelish/shortener/persistance.ex]: Do you know if it is a pattern to return a hash with {:ok, obj} or {:error, exception} in Elixir? 🤔

**Review by elainenaomi — COMMENTED** [review]
I miss the specs for this feature. 
They could improve my comprehension of the code. 🙂

**Pull request merged**